### PR TITLE
httproute: rename filter and config Go types

### PR DESCRIPTION
**Pull request by hbagdi**
Also corrects a typo.

This is a follow up to address the final comments in #249.

/cc @robscott @bowei @yiyangy

### CODE REVIEW

File: apis/v1alpha1/httproute_types.go [hunk 1]
```diff
@@ -233,7 +233,7 @@ const (
 	// header from an HTTP request before it is sent to the upstream target.
 	// Support: core
 	// +optional
-	FilterHTTPRequesttHeader = "RequestHeader"
+	FilterTypeHTTPRequestHeader = "RequestHeader"
```
Comment (robscott) [inline apis/v1alpha1/httproute_types.go:236]: Nit: I think other filter type vars should also be prefixed with FilterType as part of this change.

Comment (robscott) [reply apis/v1alpha1/httproute_types.go:236]: Another nit, it looks like the comment above this has not been updated to reflect the new var name.

Comment (hbagdi) [reply apis/v1alpha1/httproute_types.go:236]: Apologies for the dumb mistakes, PTAL.

**Comment by robscott** [comment]
Thanks!

/lgtm

**Comment by hbagdi** [comment]
Rebased to solve merge conflicts.
/assign @bowei

File: apis/v1alpha1/httproute_types.go [hunk 2]
```diff
@@ -274,14 +274,16 @@ type HTTPRouteMatch struct {
...
 	// TODO(hbagdi): add FilterHTTPRequestMirroring filter
 
-	FilterImplementationSpecific = "ImplementationSpecific"
+	// FilterTypeImplementationSpecific should be used for configuring
+	// custom filters.
+	FilterTypeImplementationSpecific = "ImplementationSpecific"
```
Comment (jpeach) [inline apis/v1alpha1/httproute_types.go:286]: Just to clarify; this constant can be used with to filter other route types?

**Comment by robscott** [comment]
Approving now so only a LGTM + rebase is required to get this in.

/approve

File: apis/v1alpha1/httproute_types.go (hunk 2 above)
Comment (hbagdi) [reply apis/v1alpha1/httproute_types.go:286]: Yes.

**Comment by hbagdi** [comment]
@robscott Rebased again.

**Comment by hbagdi** [comment]
Test should pass now.

**Pull request merged**